server/src/test/java/org/elasticsearch/transport/InboundHandlerTests.java

```diff
@@ -322,24 +318,17 @@ public void test_handshake_precedes_stream_version_for_compatibility() throws Ex
      */
     @Test
     public void test_handshake_checks_minimum_compatible_version_if_normal_version_is_not_compatible() throws Exception {
-        Version localVersion = Version.V_4_1_0;
-        NamedWriteableRegistry namedWriteableRegistry = new NamedWriteableRegistry(Collections.emptyList());
-        TransportHandshaker handshaker = new TransportHandshaker(localVersion, threadPool, (n, c, r, v) -> {});
-        TransportKeepAlive keepAlive = new TransportKeepAlive(threadPool, (c, b) -> channel.writeAndFlush(Unpooled.wrappedBuffer(b)));
-        OutboundHandler outboundHandler = new OutboundHandler("node", localVersion, new StatsTracker(), threadPool,
-            BigArrays.NON_RECYCLING_INSTANCE);
-        InboundHandler handler = new InboundHandler(threadPool, outboundHandler, namedWriteableRegistry, handshaker, keepAlive, requestHandlers,
-            responseHandlers);
-
-
+        Version localVersion = Version.V_5_0_0;
+        Version remoteVersion = Version.V_4_1_0;
```

Comment:
I replicated what we did on master: swapped remote/local as here it's failing on
 requestHeader.finishParsingHeader(requestMessage.openOrGetStreamInput());
 
```
 because of if (isRequest()) {
            if (version.before(Version.V_4_3_0)) {
```

onmaster swapping is now affected by this condition but here it is

Reply:
I've fixed the conflicts and also found an issue which only manifests here. Will update on master as well.

Reply:
Thanks!